EzMap.Api/Controllers/PoiCollectionController.cs

```diff
@@ -16,13 +17,40 @@ public class PoiCollectionController : ControllerBase
     [HttpPost("")]
     public async Task<IActionResult> AddPoiCollection([FromBody] PoiCollectionCreateDto dto,
         [FromServices] IUnitOfWork uow,
-        [FromServices] IIdentityService identityService)
+        [FromServices] IIdentityService identityService, [FromServices] IElasticSearchService elasticSearchService)
     {
-        uow.PoiCollectionRepository.AddPoiCollection(dto.WithUserId(identityService.GetUserId()));
-
-        return await uow.SaveAsync() > 0
-            ? Ok("Your poi collection is created successfully!")
-            : new StatusCodeResult(StatusCodes.Status500InternalServerError);
+        try
+        {
+            var poiCollectionId = uow.PoiCollectionRepository.AddPoiCollection(dto.WithUserId(identityService.GetUserId()));
+            var dbResult = await uow.SaveAsync();
+            if (dbResult <= 0)
+            {
+                return new StatusCodeResult(StatusCodes.Status500InternalServerError);
+            }
+            
+            var poiCollectionCreateIndex = new PoiCollectionCreateIndex(poiCollectionId, dto.Name, dto.Description);
+            var esResult = await elasticSearchService.AddOrUpdate(poiCollectionCreateIndex);
+            if (!esResult)
+            {
+                return Ok(new
+                {
+                    Message = "Your poi collection is created successfully, but indexing encountered an issue.",
```

Comment:
You don't expose your implementation details to users/public apis in error messages.

If indexing failed is not an error, do you have any way to fix the the index, or will it just stay that way? In that case your system fell in to a state where it's not consistent, some records are indexed, and some aren't, for which you don't have any way to know or to recover from.